### PR TITLE
Developed the Datastore to permanently store the comments

### PR DESCRIPTION
**Pull request by thuavu**
Developed the Datastore to permanently store the comments. Also broken down the DataServlet.java to separate the Comment class into Comment.java as well as the new comments into NewCommentServlet.java for cleaner code. In addition, cleaned up the form.html page by developing the style.css.

### CODE REVIEW

**Comment by thuavu** [comment]
image
image

**Comment by joanapascoal** [comment]
Same about PR description. Please add one that could represent a summary of what is being changed.

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.google.sps.data;
+
+/** A comment on a the history list. */
+public final class Comment {
+
+  private final long id;
+  private final String text;
+  private final long timestamp;
+
+  public Comment(long id, String text, long timestamp) {
```
Comment (joanapascoal) [inline portfolio/src/main/java/com/google/sps/data/Comment.java]: I know that the Task example use the variable name timestamp, but that it doesn't have a object meaning, so I would suggest update it to something as 'creation_time' or a similar one that you would prefer.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -14,27 +14,48 @@
...
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;         //Comment class
```
Comment (joanapascoal) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I believe that this //Comment class comment can be removed.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -14,27 +14,48 @@
...
+    for (Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId();
+        String text = (String) entity.getProperty("text");
+        long timestamp = (long) entity.getProperty("timestamp");
 
+        Comment commentNew = new Comment(id, text, timestamp);
+        comments.add(commentNew);
+
+        // Print in console to test
+        System.out.println(text);
```
Comment (joanapascoal) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Same about System.out.println

File: portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        String text = request.getParameter("text");
+        long timestamp = System.currentTimeMillis();
+
+        // Datastore
+        Entity commentEntity = new Entity("Comment");
+        commentEntity.setProperty("text", text);
+        commentEntity.setProperty("timestamp", timestamp);
+
+        // Print in console to test
+        System.out.println(text);
```
Comment (joanapascoal) [inline portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java]: Same about System.out.println

**Review by joanapascoal — APPROVED** [review]
Great work!

Just a few minor comments.

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 1 above)
Comment (thuavu) [reply portfolio/src/main/java/com/google/sps/data/Comment.java]: image
Changed `timestamp` variable to `creationTime` for code clarification.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (thuavu) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Will do.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (thuavu) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Will do.

File: portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java (hunk 4 above)
Comment (thuavu) [reply portfolio/src/main/java/com/google/sps/servlets/NewCommentServlet.java]: Will do.

**Pull request merged**